[docs] Update RTL Guide to be more clear (#13181)
mui-org/material-ui: docs/src/pages/guides/right-to-left/right-to-left.md

### 1. HTML

Make sure the `dir` attribute is set on the body, otherwise native components will break:
```html
<body dir="rtl">
```

### 2. Theme

> Set the direction on your custom theme:

Reviewer: ```suggestion
Set the direction in your custom theme:
```

```js
const theme = createMuiTheme({
  direction: 'rtl',
});
```

### 3. jss-rtl

> You need a JSS plugin to flip the styles: [jss-rtl](https://github.com/alitaheri/jss-rtl).

Reviewer: ```suggestion
You need this JSS plugin to flip the styles: [jss-rtl](https://github.com/alitaheri/jss-rtl).
```

```sh
npm install jss-rtl
```

> Even though you installed the plugin in your project our components **require** loading it to the jss instance as described below.

Reviewer: ```suggestion
Having installed the plugin in your project, Material-UI components still require it to be loaded by the jss instance, as described below.
```

> Internally, we are using this JSS plugin when `direction: 'rtl'` is set on the theme.

Reviewer: What does "internally" mean, and who is "we" in this context?

> The [CSS-in-JS documentation](/customization/css-in-js/#opting-out-of-rtl-transformation) explains a bit more about how this plugin is working. Head to the [plugin README](https://github.com/alitaheri/jss-rtl) to learn more about it.

Reviewer: ```suggestion
The [CSS-in-JS documentation](/customization/css-in-js/#opting-out-of-rtl-transformation) explains a bit more about how this plugin works. Head to the [plugin README](https://github.com/alitaheri/jss-rtl) to learn more about it.
```

> Once you have created a new JSS instance with the plugin, you need to make it available to all components in the component tree. JSS has a [`JssProvider`](https://github.com/cssinjs/react-jss) component for this:

Reviewer: ```suggestion
Once you have created a new JSS instance with the plugin, you need to make it available to all the components in the component tree. JSS has a [`JssProvider`](https://github.com/cssinjs/react-jss) component for this:
```

```sh
npm install jss-rtl
```

> Internally, we are dynamically enabling this plugin when `direction: 'rtl'` is set on the theme. Even though you installed the plugin in your project our components require loading it to the jss instance as described below.

Reviewer: Perhaps reverse the order of these sentences? 

Although TBH, the addition seems redundant to me, since it clearly states below" "Once you have created a new JSS instance with the plugin, you need to make it available to all components in the component tree." – OP just stopped reading too soon. 😄 

While you're in this part of the docs though, lets fix the Frenglish. 😉 

```diff
-explains a bit more how this plugin is working. Head to the [plugin README](https://github.com/alitaheri/jss-rtl) to know everything about it.
+explains a bit more about how this plugin works. Head to the [plugin README](https://github.com/alitaheri/jss-rtl) to learn more about it.
```

Author: ![image](https://user-images.githubusercontent.com/12185681/46731645-e9d35480-cc93-11e8-8344-4a4a0b30d6d1.png)

The word `internally` is the confusing part. I see internally I understand that's what below is an explanation to what happens internally. reading it over and over didn't make me realize I missed a whole part, but when you continue reading and it `requires` you to add jss-rtl to the JSS instance you'll then understand where you were mistaken.

Reviewer: > The word internally is the confusing part.

Even that sentence says *"when `direction: 'rtl'` is set on the theme"* though.

Would it be even clearer if it read:

"Internally, we are dynamically enabling this plugin, but *only*  when `direction: 'rtl'` is set on the theme."

Author: It is not clear that internally it is automatically used when rtl is set.
we need to find a proper explanation

Reviewer: > It is not clear that internally it is automatically used when rtl is set.

I don't see how that could be any clearer, but, lets switch it up a bit, put the onus firmly on the user:

"Internally, we dynamically enable this plugin, but only when you set `direction: 'rtl'` in the theme."

Author: But that's not happening, if that sentence is true then why should we add jss-rtl to the JSS instance manually? why isn't it internally enabled?
